contracts/protocol/bases/OfferBase.sol

```diff
@@ -309,9 +315,13 @@ contract OfferBase is ProtocolBase, IBosonOfferEvents {
         ProtocolLib.ProtocolCounters storage pc = protocolCounters();
         uint256 _startId = pc.nextExchangeId;
 
-        // Call reserveRange on voucher
         IBosonVoucher bosonVoucher = IBosonVoucher(protocolLookups().cloneAddress[offer.sellerId]);
-        bosonVoucher.reserveRange(_offerId, _startId, _length);
+
+        // _to must be the contract address or the contract owner
+        require(_to == address(bosonVoucher) || _to == bosonVoucher.owner(), INVALID_TO_ADDRESS);
```

Comment:
```suggestion
        require(_to == address(bosonVoucher) || _to == msgSender(), INVALID_TO_ADDRESS);
```

`bosonVoucher.owner()` is the assistant, who at the same time is the one calling `reserveRange`, so this is more efficient.

Probably you can msgSender to memory variable, since it's also used in `RangeReserved` event.

Reply:
@zajck but what if the assistant has transferred the voucher ownership?